dmci/distributors/file_dist.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+"""
+DMCI : File Distributor
+=======================
+
+Copyright 2021 MET Norway
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+import uuid
+import shutil
+import logging
+
+from dmci.distributors.distributor import Distributor, DistCmd
+
+logger = logging.getLogger(__name__)
+
+class FileDist(Distributor):
+
+    def __init__(self, cmd, xml_file=None, metadata_id=None, **kwargs):
+        super().__init__(cmd, xml_file, metadata_id, **kwargs)
+
+        return
+
+    def run(self):
+        """Wrapper for the various jobs, depending on command.
+        """
+        if not self.is_valid():
+            return False
+
+        if self._cmd == DistCmd.INSERT:
+            return self._add_to_archive()
+        elif self._cmd == DistCmd.UPDATE:
+            return self._add_to_archive()
+        elif self._cmd == DistCmd.DELETE:
+            logger.error("The `delete' command is not implemented")
+            return False
+
+        logger.error("Invalid command: %s" % str(self._cmd))
+
+        return False
+
+    def _add_to_archive(self):
+        """Add the xml file to the archive.
+        """
+        jobsDir = self._conf.file_archive_path
+        if jobsDir is None:
+            logger.error("No 'file_archive_path' set")
+            return False
+
+        mdID = None
+        if self._worker is not None:
+            mdID = self._worker._file_metadata_id
+        if mdID is None:
+            logger.error("No metadata_identifier provided, cannot archive file")
+            return False
+
+        try:
```

Comment:
This test should be moved to the worker class

Reply:
Hmm, the actual check is already done in the worker class. This one is just a safeguard. The return False here also goes back to the API, which should return an error to the user.